src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCell.cs

```diff
@@ -985,7 +985,17 @@ protected override void OnKeyDown(KeyEventArgs e)
                 e.Handled = true;
                 return;
             }
-            
+
+            else if(e.Key == Key.F3)
+            {
+                if (Column.CanUserSort)
+                {
+                    Column.DataGridOwner.PerformSort(Column);
+                }
+                e.Handled = true;
+                return;
```

Comment:
You should either merge the IF-case like that `else if(e.Key == Key.F3 && Column.CanUserSort)` or move these two lines within the last IF. Because if a column cannot be sorted and  a user presses F3, you still set `e.Handled` to `true` and return so the input is never send to column.